components/function-controller/internal/controllers/serverless/build_resources.go

```diff
@@ -132,6 +134,150 @@ func (r *FunctionReconciler) buildJob(instance *serverlessv1alpha1.Function, con
+
+	one := int32(1)
+	zero := int32(0)
+
+	return batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-build-", instance.GetName()),
+			Namespace:    instance.GetNamespace(),
+			Labels:       r.functionLabels(instance),
+		},
+		Spec: batchv1.JobSpec{
+			Parallelism:           &one,
+			Completions:           &one,
+			ActiveDeadlineSeconds: nil,
+			BackoffLimit:          &zero,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: r.functionLabels(instance),
+					Annotations: map[string]string{
+						"sidecar.istio.io/inject": "false",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "credentials",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: r.config.ImageRegistryDockerConfigSecretName,
+									Items: []corev1.KeyToPath{
+										{
+											Key:  ".dockerconfigjson",
+											Path: ".docker/config.json",
+										},
+									},
+								},
+							},
+						},
+						{
+							Name: "runtime",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{Name: r.config.Build.RuntimeConfigMapName},
+								},
+							},
+						},
+						{
+							Name:         "workspace",
+							VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}},
+						},
+					},
+					InitContainers: []corev1.Container{
+						{
+							Name:            "repo-fetcher",
+							Image:           "eu.gcr.io/kyma-project/function-build-init:PR-8924",
+							Env:             buildRepoFetcherEnvVars(instance),
+							ImagePullPolicy: corev1.PullAlways,
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "workspace",
+									MountPath: workspace,
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  "executor",
+							Image: r.config.Build.ExecutorImage,
+							Args:  args,
+							Resources: corev1.ResourceRequirements{
+								Limits: corev1.ResourceList{
+									corev1.ResourceMemory: r.config.Build.LimitsMemoryValue,
+									corev1.ResourceCPU:    r.config.Build.LimitsCPUValue,
+								},
+								Requests: corev1.ResourceList{
+									corev1.ResourceMemory: r.config.Build.RequestsMemoryValue,
+									corev1.ResourceCPU:    r.config.Build.RequestsCPUValue,
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								// Must be mounted with SubPath otherwise files are symlinks and it is not possible to use COPY in Dockerfile
+								// If COPY is not used, then the cache will not work
+								{Name: "workspace", MountPath: workspace},
+								{Name: "credentials", ReadOnly: true, MountPath: "/docker"},
+								{Name: "runtime", ReadOnly: true, MountPath: "/workspace/Dockerfile", SubPath: "Dockerfile"},
```

Comment:
update that comment, if it's valid only for dockerfile in this case please move it 2 lines down